torch/testing/_internal/distributed/multi_threaded_pg.py

```diff
@@ -389,13 +389,19 @@ class WorldData:
     pg_default_device: Dict[dist.ProcessGroup, torch.device]
 
 
+MTPG_TLS_KEY = "__ThreadLocalWorld"
 class ThreadLocalWorld:
     _world = threading.local()
 
     def _get_world(self) -> WorldData:
+        if torch._C._is_key_in_tls(MTPG_TLS_KEY):
+            return torch._C._get_obj_in_tls(MTPG_TLS_KEY)
```

Comment:
are we trying to ensure 2 threads (e.g. forward and autograd) receive the same object for a `_get_world()` call?  If so i am confused about how the code helps, it looks like we are allowing a TLS version of _get_world() which would allow divergence

Reply:
I'm not 100% sure because there's some bad memory management going on, I'm trying to use the same pattern as the cuda_graph code but it's not working at all for me.

Reply:
This is no longer relevant as I changed the design.